src/Interpreters/InterpreterShowSettingQuery.cpp

```diff
@@ -0,0 +1,37 @@
+#include <Interpreters/InterpreterShowSettingQuery.h>
+
+#include <Common/quoteString.h>
+#include <IO/Operators.h>
+#include <IO/WriteBufferFromString.h>
+#include <Parsers/ASTShowSettingQuery.h>
+#include <Parsers/formatAST.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/executeQuery.h>
+
+
+namespace DB
+{
+
+
+InterpreterShowSettingQuery::InterpreterShowSettingQuery(const ASTPtr & query_ptr_, ContextMutablePtr context_)
+    : WithMutableContext(context_)
+    , query_ptr(query_ptr_)
+{
+}
+
+
+String InterpreterShowSettingQuery::getRewrittenQuery()
+{
+    const auto & query = query_ptr->as<ASTShowSettingQuery &>();
+    return fmt::format(R"(SELECT value FROM system.settings WHERE name = '{0}')", query.getSettingName());
```

Comment:
This has a trivial SQL injection because the value is not escaped.
Please fix and also add a test demonstrating non-working SQL injection.

Reply:
Updated.